policies/coding-style.md

```diff
@@ -586,11 +586,59 @@ it portable enough to meet our multi-platform support goals.
 
 ## Chapter 14:  Portability
 
+### Language
+
 To maximise portability the version of C defined in ISO/IEC 9899:1990
 should be used. This is more commonly referred to as C90. ISO/IEC 9899:1999
 (also known as C99) is not supported on some platforms that OpenSSL is
 used on and therefore should be avoided.
 
+### Integer types
+
+The types `int8_t`, `uint8_t`, `int16_t`, `uint16_t`, `int32_t`, `uint32_t`,
+`int64_t` and `uint64_t` are recommended for use. These were only standardised
+in C99 but we define them ourselves if they are not available on a target
+platform.
+
+`size_t` is available. `ossl_ssize_t` should be used instead of `ssize_t` for
+portability.
+
+Most of the built-in integer types in the C language should be avoided
+or have important caveats:
+
+- Use of `long` and `unsigned long` is deprecated, as these types cannot be
+  relied upon to be larger than `int` on all platforms. Existing public APIs
+  which already use these types must be maintained, but new APIs should avoid
+  usage of them. Internal code should avoid using these types.
+
+  Use `size_t` to represent array lengths. If a signed value is needed, use
+  `ossl_ssize_t`.
+
+- Use of `short` or `unsigned short` is deprecated. In the worst case, `short` has
+  the same size as `int`.
+
+- Use `int` or `unsigned int` only where the range of values required is a subset
+  of the range `[-32767, 32767]`, as this is guaranteed by the C standard.
```

Comment:
That's wrong. We depend on 32 bit int/unsigned int and IMO we can continue to do so. I would seriously dislike to see proliferation of int32_t everywhere.

Reply:
This is not wrong.  16 bits _is_ the minimum size for ints guaranteed by the C standard.

Now, *we* have chose to require a minimum 32-bit size for int, thus basically discarding any 16-bit system (so long, DOS, you won't be missed :wink:)...  but that's not what this sentence says.